jobs/payment-jobs/tasks/cfs_create_invoice_task.py

```diff
@@ -333,6 +334,128 @@ def _create_pad_invoices(cls):  # pylint: disable=too-many-locals
+                .filter(InvoiceModel.payment_method_code == PaymentMethod.EFT.value) \
+                .filter(InvoiceModel.invoice_status_code == InvoiceStatus.APPROVED.value) \
+                .filter(InvoiceModel.id.notin_(invoice_reference_subquery)) \
+                .order_by(InvoiceModel.created_on.desc()).all()
+
+            # Get payment account
+            payment_account: PaymentAccountService = PaymentAccountService.find_by_id(eft_account.id)
+
+            if len(account_invoices) == 0:
+                continue
+            current_app.logger.debug(
+                f'Found {len(account_invoices)} invoices for account {payment_account.auth_account_id}')
+
+            # Get CFS account
+            cfs_account: CfsAccountModel = CfsAccountModel.find_effective_by_account_id(payment_account.id)
+
+            # If no CFS account then the payment method might have changed from EFT to DRAWDOWN
+            if cfs_account is None:
+                # Get last CFS account
+                cfs_account: CfsAccountModel = CfsAccountModel.query.\
+                    filter(CfsAccountModel.account_id == payment_account.id).order_by(CfsAccountModel.id.desc()).first()
+
+            # If CFS account is not ACTIVE or INACTIVE (for above case), raise error and continue
+            if cfs_account.status not in (CfsAccountStatus.ACTIVE.value, CfsAccountStatus.INACTIVE.value):
+                current_app.logger.info(f'CFS status for account {payment_account.auth_account_id} '
+                                        f'is {payment_account.cfs_account_status} skipping.')
+                continue
+
+            # Add all payment line items together
+            lines = []
+            invoice_total = Decimal('0')
+            for invoice in account_invoices:
+                lines.extend(invoice.payment_line_items)
+                invoice_total += invoice.total
+            invoice_number = account_invoices[-1].id
+            try:
+                # Get the first invoice id as the trx number for CFS
+                invoice_response = CFSService.create_account_invoice(transaction_number=invoice_number,
+                                                                     line_items=lines,
+                                                                     cfs_account=cfs_account)
+            except Exception as e:  # NOQA # pylint: disable=broad-except
+                # There is a chance that the error is a timeout from CAS side,
+                # so to make sure we are not missing any data, make a GET call for the invoice we tried to create
+                # and use it if it got created.
+                current_app.logger.info(e)  # INFO is intentional as sentry alerted only after the following try/catch
+                has_invoice_created: bool = False
+                try:
+                    # add a 10 seconds delay here as safe bet, as CFS takes time to create the invoice
```

Comment:
Shows intent, this is good the developer might not know this is a CFS limitation